learn2learn/vision/datasets/mini_imagenet.py

```diff
@@ -0,0 +1,201 @@
+from torch.utils.data import Dataset, ConcatDataset
+import pandas as pd
+from torchvision import transforms
+from PIL import Image
+import requests
+import zipfile
+import os
+import numpy as np
+import shutil
+import os
+
+
+def mkdir(dir):
+    try:
+        os.mkdir(dir)
+    except:
+        pass
+
+
+def download_file_from_google_drive(id, destination):
+    URL = "https://docs.google.com/uc?export=download"
+
+    session = requests.Session()
+
+    response = session.get(URL, params = { 'id' : id }, stream = True)
+    token = get_confirm_token(response)
+
+    if token:
+        params = { 'id' : id, 'confirm' : token }
+        response = session.get(URL, params = params, stream = True)
+
+    save_response_content(response, destination)    
+
+def get_confirm_token(response):
+    for key, value in response.cookies.items():
+        if key.startswith('download_warning'):
+            return value
+
+    return None
+
+def save_response_content(response, destination):
+    CHUNK_SIZE = 32768
+
+    with open(destination, "wb") as f:
+        for chunk in response.iter_content(CHUNK_SIZE):
+            if chunk: # filter out keep-alive new chunks
+                f.write(chunk)
+                
+
+
+class MiniImageNet(Dataset):
+    def __init__(self, subset, transform=None, target_transform=None, data_path ='./data/'):
+        """Dataset class representing miniImageNet dataset
+        # Arguments:
+            subset: Whether the dataset represents the background or evaluation set
+        """
+                 
+        file_id = '0B3Irx3uQNoBMQ1FlNXJsZUdYWEE'
+        self.data_path = data_path
+        destination_for_zip = self.data_path + '/miniImageNet.zip'
+        destination_to_extract = self.data_path + '/miniImageNet/images'
+        mkdir(self.data_path + '/miniImageNet/images_background')
+        mkdir(self.data_path + '/miniImageNet/images_evaluation')
+
+
+        self.transform = transform
+        self.target_transform = target_transform
+        if not self.transform:
+            self.transform = transforms.Compose([
```

Comment:
I am not sure if this was intentional, but shouldn't transforms always be at the users liberty?